go/vt/vtgate/shard_conn_test.go

```diff
@@ -766,4 +768,21 @@ func TestShardConnLife(t *testing.T) {
 	if s.DialCounter != 2 {
 		t.Errorf("DialCounter: %d, want 2", s.DialCounter)
 	}
+	sdc.Close()
+
+	// Do not auto-reconnect for master
+	s.Reset()
```

Comment:
This should be pulled into a separate test, something like TestReplicaShardConnLife and TestMasterShardConnLife.